feat: LambdaFunctionUrlAuth rule (#1051)
cdklabs/cdk-nag: RULES.md

| [ssm-document-not-public](https://docs.aws.amazon.com/config/latest/developerguide/ssm-document-not-public.html)                                                                 | SSM documents are not public.                                                                                                       | 1.2, 1.2.1, 1.3, 1.3.1, 1.3.2, 1.3.4, 1.3.6, 2.2.2                                                                |
| [vpc-sg-open-only-to-authorized-ports](https://docs.aws.amazon.com/config/latest/developerguide/vpc-sg-open-only-to-authorized-ports.html)                                       | The VPC Security Group restricts IPv4 TCP traffic on unauthorized ports.<sup>[2](#vpc-sg-open-only-to-authorized-ports)</sup>       | 1.2, 1.2.1, 1.3, 1.3.1, 1.3.2, 2.2.2                                                                              |

> ## Rules not in packs

Reviewer: ```suggestion
## Additional Rules

A collection of community rules that are not currently included in any of the included packs.
```

Reviewer: Adding some details here to clarify

> ## Rules not in packs

Reviewer: I think we should add a reference to the `README` about this. Perhaps under https://github.com/cdklabs/cdk-nag#available-packs section or a related section

Reviewer: @dbyod13 this needs to be addressed

Author: Apologies, missed that one. Added proposed commit now

| Rule ID            | Cause                                                                                                                                                  | Explanation                                                                                                                                                                                                                                                                                                                               |
| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | LambdaFunctionUrlAuth | The Lambda Function URL is open to the public as no authentication is implemented | AWS Lambda Function URLs allow you to invoke your function via a HTTPS end-point, setting the authentication to NONE allows anyone on the internet to invoke your function. | 

Reviewer: ```suggestion
| LambdaFunctionUrlAuth | The Lambda Function URL allows for public, unauthenticated access.  | AWS Lambda Function URLs allow you to invoke your function via a HTTPS end-point, setting the authentication to NONE allows anyone on the internet to invoke your function. | 
```

Reviewer: Taking [verbiage from the docs](https://docs.aws.amazon.com/lambda/latest/dg/urls-auth.html)
